### PR TITLE
IDS-956 implement validations in pages

### PR DESCRIPTION
**Pull request by uoa-noel**
This PR implements basic validation of user entry in the form. As part of the change, it also creates data models for Project and Drive information, and the pages now make use of the models.
Input for title, description change and custom retention period are also implemented. Member and division changes have not yet been implemented due to complexity - they should be addressed in a separate PR.
The app also reminds user of unsaved changes if they try to close the page before submission.
Some style changes are also made.

### CODE REVIEW

**Comment by JLoveUOA** [comment]
One or two little tidy up things for consistency but otherwise looks good (for my ability to test and check vue)

**Review by JLoveUOA — APPROVED** [review]
Approved but probably worth commenting out the /update members link in the router as there isn't a .vue file for it in this commit (although I suspect it will be coming in the next one)

File: web/src/views/CheckDetailsView.vue [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+function tryContinue() {
+    if (formState.areProjectDetailsCorrect === undefined){
+        document.title = "Error: " + DOCUMENT_TITLE;
+        error.value = "Select Yes if the project information is still correct.";
+    } else if (formState.areProjectDetailsCorrect) {
+            // Copy project details over as they are correct.
+            formState.project = Object.assign({}, getProject());
+            router.push("/data-classification");
+    } else {
+        router.push("/update-details");
```
[the remark this answers is not in the document]
Comment (uoa-noel) [reply web/src/views/CheckDetailsView.vue:28]: Thanks for picking this up. It's actually a bug in checking user input on the page, have fixed it now! It should prompt the user to choose a response if they haven't chosen one.

File: web/src/router/index.ts [hunk 2]
```diff
@@ -14,11 +14,21 @@ const router = createRouter({
...
+      path: "/check-details",
+      name: 'check-details',
+      component: () => import("../views/CheckDetailsView.vue")
+    },
     {
       path: "/update-details",
       name: 'update-details',
       component: () => import("../views/UpdateDetailsView.vue")
     },
+    {
```
[the remark this answers is not in the document]
Comment (uoa-noel) [reply web/src/router/index.ts]: Removed!

**Pull request merged**